clash-lib/src/Clash/Normalize/Transformations/Case.hs

```diff
@@ -283,7 +284,9 @@ caseCon' ctx@(TransformContext is0 _) e@(Case subj ty alts) = do
       (Prim pInfo,repTy:_:msgOrCallStack:_,ticks)
         | primName pInfo `elem` ["Control.Exception.Base.patError"
                                 ,"GHC.Err.undefined"
-                                ,"GHC.Err.errorWithoutStackTrace"] ->
+                                ,"GHC.Err.errorWithoutStackTrace"
+                                ,"GHC.Internal.Err.undefined"
+                                ,"GHC.Internal.Err.errorWithoutStackTrace"] ->
```

Comment:
`parError` also moved to [GHC.Internal.Control.Exception.Base](https://hackage.haskell.org/package/ghc-internal-9.1001.0/docs/src/GHC.Internal.Control.Exception.Base.html#patError).

Unless the new ghc doesn't see that as a primitive anymore.

Reply:
With all these things moving maybe now is a good time to start using TemplateHaskellQuotes to get strings like these as broadly as possible.

Reply:
I'd rather make a new issue to do that instead of doing it in this GHC 9.10 upgrade PR.

Reply:
With regards to `patError`, I guess we do not have any tests in out testsuite that trigger it for GHC 9.10. Since it is a bottoming function, however, it will not have an unfolding, and will thus end up as a primitive. I'll check the other lists of "exceptions values" that we have to see whether I need to make some updates there as well.

Reply:
I've followed your advice for the bottoming values in https://github.com/clash-lang/clash-compiler/pull/2758/commits/73a811d1157f5eb06c190c12a1857a829e0c56a9

Reply:
Yes, I agree changing everything to use TH quotes it a big change and should be a different PR.
I've been thinking for a while that doing that would be a good change as it would make the code less fragile.

It just that while looking through this PR now seen like a good time because lots of stuff has moved.
But now that you already made the PR with the new names, the best time would have been "yesterday".

Reply:
With respect to `patError`, we have [the PatError test](https://github.com/clash-lang/clash-compiler/blob/1d49cd2a528fe719beb3b2887dc440362b1b3574/tests/shouldwork/Basic/PatError.hs) that should test this.

And (using 9.10) the GHC core has a `patError`:
```
topEntity
  = \ (ds_d1di :: Maybe Int) ->
      case ds_d1di of {
        __DEFAULT ->
          case GHC.Internal.Control.Exception.Base.patError
                 @GHC.Types.LiftedRep   
                 @()
                 "tests/shouldwork/Basic/PatError.hs:6:1-22|function topEntity"#
          of {};
        Just x_a1cs -> x_a1cs
      }
```

But not in a form that clash expects, using ghc 9.0 it looked like:
```
topEntity
  = \ (ds_dLU :: Maybe Int) ->
      case ds_dLU of {
        __DEFAULT ->
          Control.Exception.Base.patError
            @'GHC.Types.LiftedRep
            @Int
            "tests/shouldwork/Basic/PatError.hs:6:1-22|function topEntity"#;
        Just x_aGO -> x_aGO
      }
```

This new form with the empty case get converted by [GHC2Core into `Clash.Normalize.Primitives.undefined`](https://github.com/clash-lang/clash-compiler/blob/1d49cd2a528fe719beb3b2887dc440362b1b3574/clash-ghc/src-ghc/Clash/GHC/GHC2Core.hs#L516-L517):

```
Clash: Compiling PatError.topEntity
PatError.topEntity before normalization:

λ(ds :: GHC.Internal.Maybe.Maybe GHC.Types.Int) ->
case ds[LocalId] of
  GHC.Internal.Maybe.Nothing  ->
    Clash.Normalize.Primitives.undefined @GHC.Types.Int
  GHC.Internal.Maybe.Just (x :: GHC.Types.Int) ->
    x[LocalId]
```